app/main/db/queries.py

```diff
@@ -288,19 +302,108 @@ def _build_consignment_filter_query(consignment_id: uuid.UUID):
                 ),
                 else_=None,
             ),
-        ),
+        ).label("closure_period"),
     )
 
     filters = [
         File.ConsignmentId == consignment_id,
         func.lower(File.FileType) == "file",
     ]
 
-    query = (
+    sub_query = (
         select.join(FileMetadata, File.FileId == FileMetadata.FileId)
         .filter(*filters)
         .group_by(File.FileId)
         .order_by(File.FileName)
+    ).subquery()
+
+    query = db.session.query(
+        sub_query.c.file_id,
+        sub_query.c.file_name,
+        func.to_char(
+            sub_query.c.date_last_modified,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("date_last_modified"),
+        sub_query.c.closure_type,
+        func.to_char(
+            sub_query.c.closure_start_date,
+            current_app.config["DEFAULT_DATE_FORMAT"],
+        ).label("closure_start_date"),
+        sub_query.c.closure_period,
     )
 
+    dt_range = get_date_range(date_range)
+    if dt_range["date_from"] is not None and dt_range["date_to"] is not None:
+        query = query.filter(
+            and_(
+                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+                >= dt_range["date_from"],
+                func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+                <= dt_range["date_to"],
+            )
+        )
+    elif dt_range["date_from"] is not None:
+        query = query.filter(
+            func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+            >= dt_range["date_from"]
+        )
+    elif dt_range["date_to"] is not None:
+        query = query.filter(
+            func.to_char(sub_query.c.date_last_modified, "YYYY-MM-DD")
+            <= dt_range["date_to"]
+        )
+
     return query
+
+
+def _build_date_range_filter_query(query, date_range):
+    dt_range = get_date_range(date_range)
+
+    if dt_range["date_from"] is not None and dt_range["date_to"] is not None:
```

Comment:
I think we can reduce some duplication by extracting the following and reusing here and for the consignment browse


```
def filter_query_by_date(date_field, date_from, date_to):
    if date_from and date_to:
        date_filter = and_(
                func.to_char(date_field, "YYYY-MM-DD")
                >= date_from,
                func.to_char(date_field, "YYYY-MM-DD")
                <= date_to,
            )
    elif date_from:
        date_filter = func.to_char(date_field, "YYYY-MM-DD") >= date_from
    elif date_to:
        date_filter = func.to_char(date_field, "YYYY-MM-DD") <= date_to

    return query.filter(date_filter)
```